### ISSUE TITLE
Energy dashboard: water support in last dev version regression with costs 

### ISSUE THREAD

**Issue by pbranly** [home-assistant/frontend#14233]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

In the last version 2022.11.0b2, water units are in m3/ft3 and results are given in m3/ft3
In the last version  20221029.0.dev - latest a very big improvement had been done . Sensors are still in m3/ft3 but results are given in L/gal.
The regression I have noticed is that prices are not any more calculated and stay at 0
I tried to come back to 2022.11.0b2 and prices are calculated 
When coming back to 20221029.dev, prices are equal to 0
I join 2 screen shots in both versions 
Phil
![7D7A639C-268F-4AB3-8A8B-F9F93CDD4B24](https://user-images.githubusercontent.com/22018509/198840494-25d60d6d-9d7e-49e8-892e-fb8090154f4c.png)
![16EAD59E-29FB-4C09-9B79-43C9C4E1E5DA](https://user-images.githubusercontent.com/22018509/198840496-1978382d-d70c-4b22-a5cd-f59dafc53886.png)

### Describe the behavior you expected

Display un L and prices calculated 

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

 20221029.0.dev - latest

### What was the last working version of Home Assistant Core?

2022.11.0b2

### In which browser are you experiencing the issue with?

All

### Which operating system are you using to run this browser?

iPadOS, windows11;

### State of relevant entities

```txt
state_class: total_increasing
unit_of_measurement: m³
device_class: water
friendly_name: Compteur eau Jour m3
```

**Comment by pbranly** [comment]
hi,
In addition in both versions beta 3 and dev, sensor in L seems and work while configuration menu asks for m3/ft3 only
Phil

**Comment by pbranly** [comment]
Hi 
L is in the version b5 but price issue is present 
Phil

**Comment by pbranly** [comment]
Issue still there in beta 6
Phil

**Comment by pbranly** [comment]
Fixed in version 2022.11.0

**Comment by pergolafabio** [comment]
Guys, how do you add water in L ? running 2023.1.2

when adding the sensor, it still asks me for m3 or gal ?

![image](https://user-images.githubusercontent.com/43251136/212159228-9a60fe3a-2ede-4fbb-98d3-c09991110ffe.png)
